frontend/javascripts/admin/dataset/dataset_upload_view.tsx

```diff
@@ -606,6 +603,22 @@ class DatasetUploadView extends React.Component<PropsWithFormAndRouter, State> {
     }
   };
 
+  isDatasetConversionEnabled = () => {
+    const uploadableDatastores = this.props.datastores.filter(
+      (datastore) => datastore.allowsUpload,
+    );
+    const hasOnlyOneDatastoreOrNone = uploadableDatastores.length <= 1;
+
+    const selectedDatastore = hasOnlyOneDatastoreOrNone
+      ? uploadableDatastores[0]
+      : this.getDatastoreForUrl(this.state.datastoreUrl);
```

Comment:
`this.state.datastoreUrl` is only set once the user submits the form. The antd form logic is quite messy, unfortunately. Before we do the setState, the url lives in the antd form object and can be retrieved via `currentFormRef.getFieldValue("datastoreUrl")`.

My suggestion would be to use `<Form ... onValuesChange={...}>` and in that handler, always set this.state.datastoreUrl. That way, both vars are synchronized. Then, the setState in the handleSubmit code does not need to set the datastoreUrl anymore.

Reply:
I applied your suggestion. Please have a look.